sklearn/linear_model/logistic.py

```diff
@@ -1198,6 +1198,14 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Returns self.
         """
+        if self.solver == 'auto':
+            self.solver = 'liblinear'
+            warnings.warn("Default solver will be changed to 'lbfgs' in 0.22",
```

Comment:
Only for L2 penalty

Reply:
Maybe instead of default say 'auto'

But you also now give the user no easy way to silence the warning. We will need a temporary setting like 'default' to manage the warning.